velox/functions/sparksql/ArraySort.cpp

```diff
@@ -278,6 +114,62 @@ std::shared_ptr<exec::VectorFunction> makeSortArray(
   }
   // Nulls are considered smallest.
   bool nullsFirst = ascending;
-  return std::make_shared<ArraySort>(ascending, nullsFirst);
+  return facebook::velox::functions::makeArraySort(
+      name,
+      inputArgs,
+      config,
+      /*ascending=*/ascending,
+      /*nullsFirst=*/nullsFirst,
+      /*throwOnNestedNull=*/false);
 }
+
+core::TypedExprPtr rewriteArraySortCall(
+    const std::string& prefix,
+    const core::TypedExprPtr& expr) {
+  auto call = asArraySortCall(prefix, expr);
+  if (call == nullptr || call->inputs().size() != 2) {
+    return nullptr;
+  }
+
+  auto lambda =
+      dynamic_cast<const core::LambdaTypedExpr*>(call->inputs()[1].get());
+  VELOX_CHECK_NOT_NULL(lambda);
+
+  // Extract 'transform' from the comparison lambda:
+  //  (x, y) -> if(func(x) < func(y),...) ===> x -> func(x).
+  if (lambda->signature()->size() != 2) {
+    return nullptr;
+  }
+
+  static const std::string kNotSupported =
+      "array_sort with comparator lambda that cannot be rewritten "
+      "into a transform is not supported: {}";
+
+  auto checker = std::make_unique<SparkSimpleComparisonChecker>();
```

Comment:
I wonder if we can add a parameter of MatcherPtr for the rewriteArraySortCall function to provide custom matcher for Presto and Spark. In this way, the other parts of code can also be reused.